### PR TITLE
Make CI action upload manuals

### PR DESCRIPTION
**Pull request by ssiccha**
Please use the following template to submit a pull request, filling
in at least the "Text for release notes" and/or "Further details".
Thank You!

# Description

## Text for release notes 

If this pull request shall **not** be mentioned in the release notes
(to be distributed in the file `CHANGES.md`),
please add the label `release notes: not needed`.

Otherwise, please proceed in one of the following ways:

- Choose a title that can serve as text for the release notes,
  and add the label `release notes: use title`.

- Put the text for the release notes **here**,
  that is, between the markers `Text for release notes`
  and `(End of text for release notes)`.

The first variant is recommended whenever the text for release notes
is suitable as title.

In both cases, please follow the style of the GAP `CHANGES.md` file
in the root directory.
In particular, please surround the names of GAP functions with backquotes.

## Further details

If necessary, please provide further details here.

# Checklist for pull request reviewers

- [ ] proper formatting

If your code contains kernel C code, run `clang-format` on it; the 
simplest way is to use `git clang-format`, e.g. like this (don't
forget to commit the resulting changes):

    git clang-format $(git merge-base HEAD master)

- [ ] usage of relevant labels

   1. either `release notes: not needed` or `release notes: to be added`
   2. at least one of the labels `bug` or `enhancement` or `new feature`
   3. for changes meant to be backported to `stable-4.X` add the `backport-to-4.X` label
   4. consider adding any of the labels `build system`, `documentation`, `kernel`, `library`, `tests`

- [ ] runnable tests
- [ ] lines changed in commits are sufficiently covered by the tests
- [ ] adequate pull request title
- [ ] well formulated text for release notes
- [ ] relevant documentation updates
- [ ] sensible comments in the code

### CODE REVIEW

**Comment by ssiccha** [comment]
I think this is finished now. Let's see whether the action succeeds.

**Comment by ssiccha** [comment]
So the makemanuals action uploads 4 files, that is the `dev`, `hpc`, `ref` and `tut` manuals:
https://github.com/gap-system/gap/runs/2038672727?check_suite_focus=true

But I think I can't see the artifacts until all jobs have finished.

**Comment by ssiccha** [comment]
I've checked the artifacts. I was able to open them all.

**Comment by wilfwilson** [comment]
This seems to work as intended. I typically prefer looking at the HTML, rather than the PDF version. Is there a particular reason not to include the `.html` files too? (and the relevant `css`/`js` files)

**Comment by ssiccha** [comment]
@wilfwilson only that I don't know which ones they are. :) I'm having a look.

**Comment by ssiccha** [comment]
I think this should do the trick. Do we definitely need the *.js files? (I have no idea :))

**Comment by wilfwilson** [comment]
I assume that we need the JS files for the HTML version of the manuals if and only if the JS files should exist at all. So: probably, yes.

I don’t have time to investigate now, but something is wrong:
```
Warning: No files were found with the provided path: doc/dev/*.{html,css,js}
doc/hpc/*.{html,css,js}
doc/ref/*.{html,css,js}
doc/tut/*.{html,css,js}. No artifacts will be uploaded.
```

**Comment by fingolfin** [comment]
I am fine with uploading the HTML, but note that the user still needs to download them and the view them in a browser; the PDF is preferable for me only because I can view it on my iPad or iPhone, unlike the HTML+js+css files. But of course offering them all may be the best solution:-)

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -189,6 +189,26 @@ jobs:
...
+            doc/tut/manual.pdf
+      - name: "Upload html manuals"
+        if: ${{ matrix.test-suites == 'makemanuals' }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: manuals-html
+          path: |
+            doc/dev/*.{html,css,js}
+            doc/hpc/*.{html,css,js}
+            doc/ref/*.{html,css,js}
```
Comment (fingolfin) [inline .github/workflows/CI.yml:210]: This didn't work - I think bash-style `{...}` patterns are not supported. But OTOH you could maybe use four patterns of the form `doc/*/*.html`

Comment (wilfwilson) [reply .github/workflows/CI.yml:210]: That kind of thing seems to work fine, see e.g. the result https://github.com/wilfwilson/gap/runs/2057072844?check_suite_focus=true of commit https://github.com/wilfwilson/gap/commit/113226445d62c82f5805d33a69594fb7c1fc9474

The log gives:
```
With the provided path, there will be 290 files uploaded
Total file count: 290 ---- Processed file #276 (95.1%)
Total size of all the files uploaded is 3459174 bytes
Finished uploading artifact manuals-html. Reported size is 3459174 bytes. There were 0 items that failed to upload
Artifact manuals-html has been successfully uploaded!
```
You'd think that would make the archive be about 3.5 MB, but it's actually 17.8 MB for some reason. Still acceptable though, I think?

Comment (ssiccha) [reply .github/workflows/CI.yml]: Just updated my PR. IMO the 3.5MB vs 17.8MB is not a problem. We only do that once per run of the CI workflow.

**Review by wilfwilson — APPROVED** [review]

**Pull request merged**